p2p/src/net/mod.rs

```diff
@@ -165,3 +182,11 @@ where
     /// Poll syncing-related event from the networking service
     async fn poll_next(&mut self) -> crate::Result<types::SyncingEvent<T>>;
 }
+
+// TODO: We can require infallible conversion after getting rid of `libp2p::MultiAddr`.
+/// Represents a fallible conversion to a bannable address.
+pub trait AsBannableAddress {
```

Comment:
Even though I don't want this to return an Option, but the question still stands, why we aren't using TryFrom trait. After removing the Option, maybe we should just go with From/Into.

Reply:
Either type (in case of `From`/`TryFrom`  any of types) or trait must be in the current crate, otherwise the compiler disallows that:
```rust
only traits defined in the current crate can be implemented for types defined outside of the crate
   --> p2p/src/net/libp2p/mod.rs:150:1
    |
150 | impl From<Multiaddr> for IpAddr {
    | ^^^^^---------------^^^^^------
    | |    |                   |
    | |    |                   `IpAddr` is not defined in the current crate
    | |    `libp2p::Multiaddr` is not defined in the current crate
    | impl doesn't use only types from inside the current crate
```